### DISCUSSION TITLE
What will break in next release

### DISCUSSION THREAD

**Discussion by vikiival** [kodadot/nft-gallery#4754] [General]
I am currently polishing last bits for minimal RMRKv2
And there would be a ton of changes:

1. Some actions were renamed

<img width="1095" alt="Screenshot 2023-01-18 at 13 57 23" src="https://user-images.githubusercontent.com/22471030/213177469-15e5f923-ab7c-4925-b87c-dddeeee24fd9.png">

`MINTNFT` -> `MINT`
`CONSUME` -> `BURN`

**Comment by yangwao** [comment]
Can we not merge this update till we push out this one to `main` & `hyper-jpeg` 
- https://github.com/kodadot/nft-gallery/issues/4437

**Reply by vikiival** [reply]
Yes will update also snek with suitable data

**Comment by vikiival** [comment]
We need to find a way how to unify metadata for chains

How Kodadot works?

We use OpenSea metadata standard so everything which was minted through us will is following this.

What others have you seen?

Example of metadata

```json
{
	"name": "Kodachain - Bled Mini-conference 2023",
	"image": "ipfs://ipfs/bafybeiasrjssdtaqegmsx4nuhcxnjoy6iw6pyeou3l7qf7gp7fvgst6l3y",
	"animation_url": "ipfs://ipfs/bafybeievjhx7fxkgrkgyktbmominjy7u2f5qhcl6decspyhnxhq7hnumhe",
	"description": "Congratulations! You are the owner of this commemorative Kodachain. You received it for attending the event on the 6th of May 2023. This event took place in SubWork Bled (Slovenia) and it was full of interesting speakers mostly from the Polkadot ecosystem. Created by [Sebicified](https://twitter.com/sebicified)./n Participate and join our future events so you can collect more Kodachains!"
}
```

- https://github.com/kodadot/nft-gallery/pull/6080
- https://github.com/kodadot/nft-gallery/pull/6956
- https://github.com/kodadot/workers/issues/95

**Comment by vikiival** [comment]
@kodadot/internal-dev: we plan to release a squid versions with unified addresses (no more different addresses per chain).

<img width="641" alt="Screenshot 2023-08-30 at 17 09 23" src="https://github.com/kodadot/nft-gallery/assets/22471030/d01ac8f2-eb8a-4ff3-8871-4efa5570b242">


Question is which address format is more friendly for you:
1. Polkadot (prefix 0) Vote ❤️ 
2. General substrate (prefix 42) vote 🚀 

@kodadot/design-squad, @kodadot/internal-dev, @kodadot/internal  please vote (for 1. ❤️, for 2. 🚀 )

**Reply by roiLeo** [reply]
🎉 prefix 2?

**Reply by vikiival** [reply]
Kusama adresses are sexy, I know, but

<img width="544" alt="Screenshot 2023-08-30 at 17 22 39" src="https://github.com/kodadot/nft-gallery/assets/22471030/71e98915-3fa2-4a38-a53b-6f1775822407">

That's why I narrowed it for two options

**Reply by JustLuuuu** [reply]
What is easier for users to understand? Also for newcomers (peeps outside of ecosystem)? In Nova Wallet, I tried to find my generic one and I was not able to find it. So I would say Polkadot

**Reply by vikiival** [reply]
> What is easier for users to understand?

That is a very good and valid question: 
1. React fork of KodaDot is using (42)
2. I honestly think many newcomers come via polkadot so they see (prefix 0)

That is the reason I started the discussion to collectively decide.

**Reply by jsidorenko** [reply]
Developers usually use 42, since this is what set locally + on westend and rococo

**Reply by damskyftw** [reply]
If something it should be Polkadot :)

**Reply by vikiival** [reply]
Since #10354 is up I am in favour of use Substrate generic addresses.

**Reply by vikiival** [reply]
- #3019

**Comment by vikiival** [comment]
# Dev update 09/23 
 
## 1. SubSquid pricing

[As you may know, subsquid is going for a paid model](https://docs.subsquid.io/deploy-squid/pricing/) when we are billed for indexing, data storage, and API availability. 
 This pricing is very friendly; however, we can still save money on the infrastructure. 
 Therefore, we decided to do a few optimizations:

### 1. Deprecation of old squids
 
 since SubSquid cannot bill old squids, they will have end-of-life on 1st October 2023.
 Namely: click, antick, rubick@008
 There is a WIP to bring the moonbeam indexer again 

- #7369
- #7373

Edit:
 There is an unknown bug in rubick/marck #7373 that they index less nfts as they should because of this reason I removed rubickV9, forked 008 and released v8 which will be supported for the longer period of time.

 
<img width="641" alt="Screenshot 2023-09-28 at 15 45 13" src="https://github.com/kodadot/nft-gallery/assets/22471030/fce0ce84-5f2c-438b-9d0b-b79fbc6bc694">



### 2. Removing snek chain
 
 Sadly, as it was our only test indexer, we no longer plan to support Snek, as the bill for that is much more than developers usually pay for testing on AHK. 
 Is there going to be a testnet? There is an issue with creating AHW or AHR (Westend or rococo) for devs to try all the features. 
 Contact me if you are familiar with deploying docker images to some pricing/dev-friendly infra. We can pay a bounty for the setup.
 EOL is also on 1st October 2023
 
Edit:
 I managed to configure snek into very minimal squid so I would move Its life for at least 1st Nov 2023
 
 
<img width="1223" alt="Screenshot 2023-09-28 at 15 45 05" src="https://github.com/kodadot/nft-gallery/assets/22471030/e9c5ea7f-ebfe-4e34-84e7-51dbecc641cd">

As I mentioned I am still in favour of running AHR (rococo) for the test purposes.


### 3. Combining RMRK + KSM
 
 According to pricing, the part of “indexing” is the highest on the bill.
 Since rmrk and KSM index the same chain, they could be merged under “the one” indexer. For this change to happen, there is currently no date to make it happen. See parts 2.1 and 2.2

Edit: Most probably this won't happen as addressed in edit of 1.1

## 2. Breaking changes

### 1. Unified addresses

As you have seen on kodadot.xyz, the address of a logged-in user changes based on the chain in which they are logged in. This is, however, sad for UX as the user always has to “learn” his/her address.
As agreed per the discussion above, we will show the address in the Polkadot prefix on the subsquid side; however, we plan to use the original ss58 address as it is more common in the ecosystems, and both wallets prefer them by default.

<img width="398" alt="Screenshot 2023-09-26 at 11 33 45" src="https://github.com/kodadot/nft-gallery/assets/22471030/5a4010a0-5efa-48fb-ab9e-6a10d626846e">

<img width="535" alt="Screenshot 2023-09-26 at 11 32 57" src="https://github.com/kodadot/nft-gallery/assets/22471030/766249c4-f02e-432a-86be-412883a356d4">


### 2. Prefixed legacy nfts
 
 To support two protocols per squid, we decided to prefix nfts and collection with the prefix u- as it stands for (uniques old pallet on asset hub ). As mentioned in 1.3, a similar thing will come to the rmrk1.0, where nfts will get the same prefix. This is just an aware note to construct the correct extrinsic.
 

## 3. Data availability / Workers

Apart from my assigned tasks, my current focus will be Squids and workers. If you find some parts of the app slow, please open an issue in Kodadot/loligo, ideally accompanied by another issue from nft-marketplace

**Reply by vikiival** [reply]
Possibly found [render.com](https://render.com/pricing#compute)

**Reply by preschian** [reply]
> Sadly, as it was our only test indexer, we no longer plan to support Snek

sad, I like `snek`. Because testing on mainnet is quite expensive

**Reply by daiagi** [reply]
> sad, I like snek. Because testing on mainnet is quite expensive

@preschian  do you turn off the support fees when testing?

**Reply by preschian** [reply]
@daiagi oh no, I keep it on. is this under the settings menu?

**Reply by daiagi** [reply]
yep. it isn't expensive if you turn it off
![image](https://github.com/kodadot/nft-gallery/assets/22791238/6d8d039e-ed1f-4532-8659-0cfa0ec96e1a)

**Reply by preschian** [reply]
aahh I see, thank you so much, I will turn it off now

**Reply by vikiival** [reply]
> sad, I like `snek`. Because testing on mainnet is quite expensive

I will try to optimize the squid pricing so we can keep it running for testing. 
I am still in favour of switching to ahw/ahr.

**Reply by preschian** [reply]
> I am still in favour of switching to ahw/ahr

this one is also better 👍

**Reply by exezbcz** [reply]
> 1. Unified addresses

this is good!

**Reply by vikiival** [reply]
Updated ^-^ with edits
